src/librustc_mir/transform/const_prop.rs

```diff
@@ -192,7 +192,10 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for ConstPropMachine {
         _ret: Option<(PlaceTy<'tcx>, BasicBlock)>,
         _unwind: Option<BasicBlock>,
     ) -> InterpResult<'tcx> {
-        throw_unsup!(ConstPropUnsupported("calling intrinsics isn't supported in ConstProp"))
+        #[derive(Debug)]
+        struct ConstPropIntrinsic;
+        impl MachineStopType for ConstPropIntrinsic {}
```

Comment:
I considered writing a macro to avoid repeating this... but that seemed like overkill?
OTOH, then I could also add the boilerplate to avoid the `unsafe`.

Reply:
Okay the `unsafe` is gone. I could still have a macro that looks like
```
throw_machine_stop_str("foo bar");
```
and internally does everything: create a ZST struct, implement `Debug` for it with message `"foo bar"`, and then throw it as a machine stop error.

Reply:
That seems like a neat idea

Reply:
r=me either with or without this change

Reply:
The idea was too neat to resist. ;)